
## Model Organisms
### Yeast
**Saccharomyces cerevisiae** is a well-studied eukaryote that grows like a prokaryote. Only 4% of the genes have introns and there is very little intergenic space. ORF finding works well to identify genes. [NCBI Genome Data Viewer](https://www.ncbi.nlm.nih.gov/genome/gdv/?org=saccharomyces-cerevisiae) and [UCSC](http://genome.ucsc.edu/cgi-bin/hgGateway?hgsid=155744526&clade=other&org=0&db=0) are good general resources for s. cerevisiae. [SDG](http://genome.ucsc.edu/cgi-bin/hgGateway?hgsid=155744526&clade=other&org=0&db=0) is an extensive resource for budding yeast genomes, including genome browsing, info on ORFs and protein domains, 3D structures, BLAST, ontology terms, and gene expression data. [ScerTF](http://stormo.wustl.edu/ScerTF) is a database of s. cerevisiae transcription factors. 

**Gene Duplication** occurs often in eukaryotes, with 4 broad fates after gene duplication: both copies remain w/ little mutation providing extra source for transcription/gene expression; one copy is deleted or permanently damaged; one copy accumulates mutations & becomes a pseudogene; and paralogous - one or both copies evolve to have different functions. The deletion fate is most common. The [Yeast Gene Order Browser](http://ygob.ucd.ie/) is a tool for visualising the syntenic context of any gene from several yeast genomes.

**Fission Yeast** Schizosaccharomyces pombe better resembles higher eukaryotes than s. cerevisiae due to the increased presence of introns. [PomBase](https://www.pombase.org/) is a genome browser that includes s. pombe and other species in the Schizosaccharomyces genus. 

**Aspergillus** resources include [AspGD](http://www.aspgd.org/), an Aspergillus genome database, and [Aspergillus Comparative Database](https://www.broadinstitute.org/fungal-genome-initiative/aspergillus-genome-projects) which focuses on comparative genomics in Aspergillus. A good **Candida** resource is [Candida Genome Database](http://candidagenome.org/), which has manually curated sequence data for Candida species. For general **Fungi** resources, there is the data-mining interface [FungiDB](https://fungidb.org/fungidb/app) and the genome browser [EnsemblFungi](http://fungi.ensembl.org/index.html).

### Non-Metazoan
The [Sanger Protozoan Genomes Projects](https://www.sanger.ac.uk/resources/downloads/protozoa/) provides access to protozoan genomes sequenced at the Wellcome Sanger Institute. Other more specific databases include [Giardia DB](https://giardiadb.org/giardiadb/app), Trypanosomes & Leishmania via [GeneDB](https://www.genedb.org/), and [Trypanocyc](http://vm-trypanocyc.toulouse.inra.fr/).

**Plasmodium** can be accessed via [PlasmoDB](https://plasmodb.org/plasmo/app), **Tetrahymena** can be accessed via [Tetrahymena DB](http://ciliate.org/index.php/home/welcome) and are known for their abundance of telomeres, and **Toxoplasma gondii** can be accessed via [ToxoDB](https://toxodb.org/toxo/app).

**Thalassiosira pseudonana** can be accessed via [Thalassiosira at JGI](https://mycocosm.jgi.doe.gov/Thaps3/Thaps3.home.html) and is believed to have acquired a plastid by engulfing a eukaryotic alga. **Phytophthora** can be accessed via [MycoCosm](https://mycocosm.jgi.doe.gov/Phyra1_1/Phyra1_1.home.html).

### Plants
**Arabidopsis thaliana** is the most extensively studied multicellular plant. Databases include [TAIR](https://www.arabidopsis.org/) and the more general [PlantsDB](http://pgsb.helmholtz-muenchen.de/plant/plantsdb.jsp). [IRGSP](https://rgp.dna.affrc.go.jp/IRGSP/) is the database for **Oryza sativa** (rice) and [MaizeGDB](https://www.maizegdb.org/) for **Zea mays** (corn). **Dictyostelium discoideum**, a migrating slime mold, is the closest unicellular organism to Metazoans, and can be accessed via [dictyBase](http://dictybase.org/)

### Metazoan Genomes
**Ciona intestinalis** is the smallest of any experimentally manipulable chordate and provides a good system for exploring the evolutionary origins of the chordate lineage. [Ciona](https://mycocosm.jgi.doe.gov/Cioin2/Cioin2.home.html) via MycoCosm is a good database. 

**Caenorhabditis elegans** is a roundworm model system. RNAi came from the study of c. elegans, and they are used to examine the timing of developmental events. [Helminth](http://helminth.net/HN_frontpage.cgi) and [WormBase](https://wormbase.org//#012-34-5) are good resources for c. elegans. 

**Drosophila melanogaster** has been a model system used to study developmental patterns. 30% of d. melanogaster genes have a worm ortholog and 20% have a yeast ortholog. These candidate genes are believed to be necessary for eukaryotic life. [FlyBase](http://flybase.org/) is a good fly genomics resource. 

For higher organisms, Ensembl, UCSC, and IGV are good options. There is also a **Mouse** database, [Mouse Genome Informatics](http://www.informatics.jax.org/), and a **Chimpanzee** database, [Chimpanzee Genome Project](https://www.broadinstitute.org/chimpanzee/chimpanzee-genome-project).
