## NGS Sequencing and Analysis: RNA and Beyond

### RNA-seq
RNA-seq is the term for massively parallel cDNA sequencing. The benefits of RNA-seq include defining transcription start sites, revealing small noncoding RNAs and long noncoding RNAs, identifying alternative splice products and splice sites, and quantitatively estimating transcript levels.

Alignment of RNA-seq data to a reference genome presents some unique challenges, specifiically with aligning spliced RNA to genomic DNA and accounting for gaps (introns) appropriately. Specialized software like [TopHat](http://ccb.jhu.edu/software/tophat/manual.shtml) is required for RNA-seq analysis. TopHat consists of [Cufflinks](http://cole-trapnell-lab.github.io/cufflinks/cufflinks/index.html#cufflinks-output-files) - uses a probabilistic model to take assembled reads and output complete transcripts; [Cuffcompare](http://cole-trapnell-lab.github.io/cufflinks/cuffcompare/) - compares two or more GFT files from Cufflinks, and [Cuffdiff](http://cole-trapnell-lab.github.io/cufflinks/cuffdiff/index.html#cuffdiff-output-files) - does the statistical analysis of differential expression between the two datasets. [Cuffmerge](http://cole-trapnell-lab.github.io/cufflinks/cuffmerge/index.html#cuffmerge-output-files) is also useful for producing a merged transcript dataset that includes all transcripts in both datasets, and [CummeRbund](http://compbio.mit.edu/cummeRbund/index.html) is an R package that is designed to aid and simplify the task of analyzing Cuffdiff RNA-Seq output. 

Other RNA-seq analysis tools inlcude: [MATS](https://pubmed.ncbi.nlm.nih.gov/22266656/) designed for hypothesis testing of alternative splice patterns detected by RNA-seq; [DiffSplice](https://pubmed.ncbi.nlm.nih.gov/23155066/) compares transcriptomes assembled from multiple RNA-seq datasets; [DESeq2](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8) performs differential expression testing of genes from RNA-seq data; [edgR](https://academic.oup.com/bioinformatics/article/26/1/139/182458), [EBSeq](https://www.bioconductor.org/packages/devel/bioc/vignettes/EBSeq/inst/doc/EBSeq_Vignette.pdf), and [baySeq](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-422) rely on first aligning RNA-seq reads to a reference genome before transcript abundances can be estimated; and [StringTie](https://ccb.jhu.edu/software/stringtie/) performs transcript assembly and expression estimates in one step.

There are also de novo RNA-seq analysis tools which are useful in organisms that do not have a well-annotated genome. They work by aligning RNA-seq reads to each other based on overlapping segments to recapitulate a full RNA molecule. [Trinity](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3571712/), [Salmon](https://salmon.readthedocs.io/en/latest/), [Sailfish](https://www.nature.com/articles/nbt.2862), [RSEM](https://github.com/deweylab/RSEM), [Oases](https://pubmed.ncbi.nlm.nih.gov/22368243/), and [Kallisto](https://www.nature.com/articles/nbt.3519) are de novo assembly tools or pseudo-mappers. 

An example RNA-seq analysis might be: *FastQC* to check data quality -> *Trimmomatic* to trim reads -> *TopHat* to map reads to reference -> *Cufflinks* to assemble reads into transcripts -> *Cuffmerge* to merge multiple datasets with reference annotation -> *Cuffdiff* to perform differential expression analysis.

### ChIP-seq
Two copies of each of four proteins make up a **Histone** octomer. Histones associate with DNA, and changes in which regions of DNA associate with DNA have implications for transcription, regulation, and expression. Fene expression regulation by histone variants has been linked to cellular plasticity, particularly during development. There are several variants of some histone proteins, and in some celllular conditions some histone variants may be favored to alter the makeup of the octomers. Additionally, **Histone Modifications** are post-translational modification of individual amino acids in the histone proteins. A few well-known histone modifications that impact gene expression include _methylation, acetylation, phosphorylation, sumoylation, ubiquitination, and ADP ribosylation_. A simplified version of the histone modification patterns associated with activated and repressed genes is shown below. 
![histone.png](/files/histone.png)
immunoprecipitation is 
chromatin immunoprecipitation is
chromatin immunoprecipitation (ChIP) procedure is used to precipitate both indented histone protein and the DNA it is bound to. It can be used for the purposes of sequencing the bound DNA to determine what DNA regions associate with the protein of interest.

At a high level, ChIP-seq consists of precipitating protein-DNA complexes, sequencing the DNA, aligning the reads where a pileup of reads is interpreted as associated with the histone, and using negative controls to remove background binding. The three types of contol experiements are DNA input control, IgG antibody control, and the use of untagged strains. 


### Single-Cell RNA-seq
