## NGS Sequencing and Analysis: DNA
### Reference Genome and Sequencing Technologies
A **reference genome** (also known as a reference assembly) is a digital nucleic acid sequence database, assembled as a representative example of the set of genes in one idealized individual organism of a species. The most recent human reference genome, GRCh38, is derived from thirteen anonymous volunteers.

Next-Generation (Second Generation) Sequencing is characterized by ultra-high throughput and massively parallel sequencing, and the need to prepare amplified sequencing banks before starting the sequencing of amplified DNA clones. These include [Illumina](https://www.illumina.com/science/technology/next-generation-sequencing/sequencing-technology.html), [ABI SOLiD](https://www.youtube.com/watch?v=nlvyF8bFDwM&ab_channel=KamleshSahu), and [Ion Torrent](https://www.youtube.com/watch?v=zBPKj0mMcDg&ab_channel=ThermoFisherScientific). Third Generation Sequencing technologies are classified as single molecule sequencing, and have the capability to produce substantially longer reads with higher error rates. Some examples include [Pacific Biosciences](https://www.pacb.com/) and [Oxford Nanopore](https://www.youtube.com/watch?v=GUb1TZvMWsw&t=17s&ab_channel=OxfordNanoporeTechnologies).

The [Sequence Read Archive](https://www.ncbi.nlm.nih.gov/sra) stores original, unprocessed sequencing data generated as part of published projects and studies. It was designed to prevent the loss of original sequencing data.
### Genome Sequence Assembly
**Phred Quality Scores** were established during the HGP and tell you how confident the call is for each nucleotide sequenced. Quality scores are important in NGS data analysis, and are encoded in FASTQ files. 

At a high level, genome sequence assembly consists of assembling reads into contigs, and contigs into full chromosomes. **De Novo Assembly** is the process of aligning a genome from scratch usually through the use of a graphical approach, either an overlap graph, de Bruijn graph, or string graph. [Velvet](https://pubmed.ncbi.nlm.nih.gov/18349386/) is an example of a de novo assembler that uses de Bruijn graph approach to generate contigs.  The appropriate de novo assembly package depends on a number of factors, particularly single-end vs paired-end, organism, and sequencing platform as mentioned in this [de novo assemblers review](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5826002/). **Reference-Based Assembly** uses a reference genome to guide the alignment process. [Bowtie](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2009-10-3-r25) uses the Burrows-Wheeler indexing approach and is incredibly fast. [BWA](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2705234/) also uses the Burrows-Wheeler indexing approach. [HISAT](https://www.nature.com/articles/nmeth.3317) is the newest invocation of the Burrows-Wheeler alignment, with a low memory footprint and built-in ability to interface with downstream analysis steps. 
### Variant Calling
The most likely explanations for a mismatch with the reference sequence are sequencing error, heterozygous SNP, homozygous SNP, and PCR amplification errors. _Sequencing Error_ is when the wrong base was called during sequencing. The rate and types of errors depend on the technology used. Only one or very few reads will have the error, and deeper library sequencing will result in higher coverage at a given locus which helps drown out the sequencing error. A _Heterozygous SNP_ is when ~50% of the reads map to the reference due to a SNP, while a _Homozygous SNP_ is when all reads differ from the reference due to a SNP. _PCR Amplification Errors_ are errors introduced during library amplification prior to sequencing. They are the hardest to distinguish from variants, as a single error early on will propagate and create many copies of the error.

The goal of variant calling is to determine if a mismatch is due to a SNP or an error. Better coverage makes variant calling easier because more reads = more confidence. Five known variant callers are [FreeBayes](https://arxiv.org/abs/1207.3907v2), [SAMtools](http://www.htslib.org/doc/samtools.html), [GATK Unified Genotyper](https://gatk.broadinstitute.org/hc/en-us), [GATK Haplotype Caller](https://gatk.broadinstitute.org/hc/en-us), and [Platypus](https://www.well.ox.ac.uk/platypus). Variant callers take an input of an aligned BAM file and a reference genome, and typically output a VCF. Variant packages like [SnpEff](https://pcingola.github.io/SnpEff/) and [ANNOVAR](https://annovar.openbioinformatics.org/en/latest/) predict functional impact from a VCF file. 

An example variant calling workflow is _FASTQ Groomer_ to reformat FASTQ files -> _FASTQ Trimmer or Trimmomatic_ to trim low quality reads -> _Bowtie2/BWA/HISAT_ to align reads in FASTQ file to a reference genome -> _SAM-to-BAM_ if necessary -> _FreeBayes_ to produce a VCF of variants 

### Metagenomics
Metagenomics is the study of genetic material obtained from environmental samples. Generally, NGS is performed on a sample and the reads are aligned to potential reference genomes to detect matches. Publically available tools for the analysis of metagenomes include [Real-Time Metagenomics](http://edwards.sdsu.edu/rtmg/), [Rapid Annotation using Subsystem Technology (RAST)](https://rast.nmpdr.org/), [Pipeline Analysis for Next Generation Amplicons (PANGEA)](https://sourceforge.net/projects/pangea-16s/), [Metagenomics tutorials in Galaxy](https://galaxyproject.github.io/training-material/topics/metagenomics/), and [Metagenome Analyzer (MEGAN)](https://www.wsi.uni-tuebingen.de/lehrstuehle/algorithms-in-bioinformatics/software/megan6/).

### SNPs
A polymorphism involves one of two or more variants of a particular DNA sequence, with some definitions requiring the least common allele to have a frequency of about 1% or greater. A SNP is a substitution of a single nucleotide at a specific position in the genome. 

The functional consequences of a SNP include _coding_ - protein variation; _silent_ - doesn't impact protein coding but may impact splicing; _promoter or regulatory_ - may impact expression; and _other_ - no known impact, useful as a chromosomal marker. 

The three categories of SNPs are _synonymous_ - coding region, amino acid doesn't change; _non-synonymous_ - coding region, amino acid changes; and _noncoding_ - in a noncoding region.

A number of SNP databases exist. [dbSNP](https://www.ncbi.nlm.nih.gov/snp/) via NCBI is the largest database of SNPs, MNPs, and short indels. [SNPEffect](https://snpeffect.switchlab.org/) focuses on predicting the impact of SNPs on the structural phenotype of proteins. [SNPedia](https://www.snpedia.com/index.php/SNPedia) is a wiki that collects information from publications on known SNPs and their clinical effects. The [1000 Genomes Browser](https://www.ncbi.nlm.nih.gov/variation/tools/1000genomes/) and [Variation Viewer](https://www.ncbi.nlm.nih.gov/variation/view/) are good resources for visualizing SNPs. [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/) is a curated database of variants and their phenotypic or clinical relevance.

### CNVs
Copy number variation and other structural variations have been identified in human, chimp, fly, and mouse genomes. The microhomology-mediated break induced replication ([MMBIR](https://pubmed.ncbi.nlm.nih.gov/19597530/)) method may explain where there seem to be hotspots in genomes for CNVs. Non-allelic homologous recombination ([NAHR](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0633-1)) is also linked to copy number variations, deletions, and inversions. CNVs can be neutral or linked to disease.

The [Database of Genomic Variants](http://dgv.tcag.ca/dgv/app/home) has a DGV Structural Variants track to visualize CNVs, which is also found on UCSC. Other CNV databases include [DECIPHER](https://www.deciphergenomics.org/) - the DatabasE of Chromosomal Imbalance and Phenotype in Humans, [Variation Viewer](https://www.ncbi.nlm.nih.gov/variation/view/) - exon pattern track and dbVar ClinVar Large Variations track, [gnomAD SVs](https://gnomad.broadinstitute.org/region/1-55505221-55530525?dataset=gnomad_sv_r2_1) - visualize structural variants in the population, and NCBI's [dbVAR](https://www.ncbi.nlm.nih.gov/dbvar/). 



