# ðŸ§¬-ðŸ’»-ðŸŒŸ Tools for Genome Analysis
_Sara Carsanaro's Final Portfolio for John Hopkins Spring 2021 Course AS.410.635 Bioinformatics: Tools for Genome Analysis_

## [Genomic Databases](./genomic_databases.md)
- databases including Ensembl, Biomart, UCSC Table Browser, and ENCODE
- web applications including UCSC and Galaxy (move do DNA seq analysis)
- desktop applications including IGV

## [NGS Sequencing and Analysis: DNA](./NGS_sequencing_analysis.md)
- SNVs and CNVs
- BEDtools and SAMtools
- de novo and reference based sequence assembly
- variant calling

## [NGS Sequencing and Analysis: RNA and Beyond](./NGS_RNA.md)
- ChIP-seq, RNA-seq, single-cell RNA-seq
- CRISPR for gene oncology
- introduction to model organisms

## [Gene Annotation and Structure](./genome_annotation.md)
- prokaryotic and eukaryotic gene structure
- gene finding programs

## [Genomic File Formats](./file_formats.md)
- various file formats you may encounter working in NGS workflows

## [Homework and Exams](./files/all_files.md)
- linked homework and exams from the course


need to include somewhere
- noncoding NRAs and UCRs

```markdown
Syntax highlighted code block

# Header 1
## Header 2
### Header 3

- Bulleted
- List

1. Numbered
2. List

**Bold** and _Italic_ and `Code` text

[Link](url) and ![Image](src)
```

For more details see [GitHub Flavored Markdown](https://guides.github.com/features/mastering-markdown/).

